### PR TITLE
WIP of converting examples to Webpack v2

### PR DESCRIPTION
**Pull request by justin808**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/742)

### CODE REVIEW

**Comment by justin808** [comment]
@alexfedoseev Can you take a look at my conversions of the webpack configs in `/spec/dummy/client`.

File: spec/dummy/client/.babelrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "presets": ["es2015", "stage-0", "react"],
+  "presets": ["es2015", "stage-2", "react"],
   "plugins": ["syntax-decorators"]
```
Comment (alex35mil) [inline spec/dummy/client/.babelrc]: I don't think it should be here.

File: spec/dummy/client/webpack.client.rails.build.config.js [hunk 2]
```diff
@@ -19,43 +16,86 @@ config.output = {
...
   {
     test: /\.css$/,
-    loader: ExtractTextPlugin.extract(
-      'style',
-      'css?minimize&modules&importLoaders=1&localIdentName=[name]__[local]__[hash:base64:5]' +
-      '!postcss'
-    ),
+    loader: ExtractTextPlugin.extract({
+      fallback: 'style-loader',
+      loader: [
```
Comment (alex35mil) [inline spec/dummy/client/webpack.client.rails.build.config.js]: `use:`

File: spec/dummy/client/webpack.client.rails.build.config.js [hunk 3]
```diff
@@ -19,43 +16,86 @@ config.output = {
...
+        {
+          loader: 'css-loader',
+          query: {
```
Comment (alex35mil) [inline spec/dummy/client/webpack.client.rails.build.config.js]: `options:`

File: spec/dummy/client/webpack.client.rails.build.config.js [hunk 4]
```diff
@@ -19,43 +16,86 @@ config.output = {
...
-    loader: ExtractTextPlugin.extract(
-      'style',
-      'css?minimize&modules&importLoaders=3&localIdentName=[name]__[local]__[hash:base64:5]' +
-      '!postcss' +
-      '!sass' +
-      '!sass-resources'
-    ),
+    loader: ExtractTextPlugin.extract({
+      fallback: 'style-loader',
+      loader: [
```
Comment (alex35mil) [inline spec/dummy/client/webpack.client.rails.build.config.js]: `use:`

File: spec/dummy/client/webpack.client.rails.build.config.js [hunk 5]
```diff
@@ -19,43 +16,86 @@ config.output = {
...
+        {
+          loader: 'css-loader',
+          query: {
```
Comment (alex35mil) [inline spec/dummy/client/webpack.client.rails.build.config.js]: `options:`

File: spec/dummy/client/.babelrc (hunk 1 above)
Comment (justin808) [reply spec/dummy/client/.babelrc]: you mean syntax-decorators?

Comment (alex35mil) [reply spec/dummy/client/.babelrc]: Yeah, AFAIR we don't use them anywhere.

**Comment by justin808** [comment]
Still some issues:

1. Tests fail with no JS being run due to a syntax error. See below. `cd spec/dummy && rspec`.
2. Hot reloading is broken with this error below. `foreman start -f Procfile.hot`


### Phantomjs issue
<details>
➜  ~/shakacode/react-on-rails/react_on_rails/spec/dummy (update-to-webpack-v2 u=) ✗ rspec './spec/features/integration_spec.rb[1:1:1:1]'                                                                                       ✹ [21:00:16]
Capybara using driver: poltergeist
Run options: include {:ids=>{"./spec/features/integration_spec.rb"=>["1:1:1:1"]}}

Pages/Index
  All in one page
    Server Rendered/Cached React/Redux Component

====> React On Rails: Checking app/assets/webpack for outdated/missing bundles

[SERVER] RENDERED RenderedHtml to dom node with id: HelloWorld-react-component-4 with railsContext: {"inMailer":false,"i18nLocale":"en","i18nDefaultLocale":"en","href":"http://127.0.0.1:59126/","location":"/","scheme":"http","host":"127.0.0.1","port":59126,"pathname":"/","search":null,"httpAcceptLanguage":"en-US,*","somethingUseful":"REALLY USEFUL","serverSide":true}
TURBO: USING TURBOLINKS 2: document added event listeners page:before-unload and page:change.
TURBO: reactOnRailsPageLoaded
RENDERED ReduxApp to dom node with id: ReduxApp-react-component-0 with props, railsContext: [object Object] [object Object]
      example at ./spec/features/integration_spec.rb:15 (FAILED - 1)
  HTML screenshot: file:///Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/tmp/capybara/screenshot_2017-03-01-21-00-44.888.html
  Image screenshot: file:///Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/tmp/capybara/screenshot_2017-03-01-21-00-44.888.png

Failures:

  1) Pages/Index All in one page Server Rendered/Cached React/Redux Component
     Failure/Error: visit root_path

     Capybara::Poltergeist::JavascriptError:
       One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).

       TypeError: ReactOnRails encountered an error while rendering component: ReduxApp.
       Original message: undefined is not a constructor (evaluating 'Object.assign({}, props, { railsContext: railsContext })')
       TypeError: ReactOnRails encountered an error while rendering component: ReduxApp.
       Original message: undefined is not a constructor (evaluating 'Object.assign({}, props, { railsContext: railsContext })')
           at undefined:10 in default
           at :41 in default
           at :45 in createReactElement
           at :133 in render
           at :66 in forEach
           at :71 in forEachComponent
           at :167 in reactOnRailsPageLoaded
           at :0 in dispatchEvent
           at http://127.0.0.1:59126/assets/application_static-b5e9b7937a49f12d63506a83dcc03e6b8a0f635b24e2cc91ed1234c6688e6fb7.js:4572 in triggerEvent
           at http://127.0.0.1:59126/assets/application_static-b5e9b7937a49f12d63506a83dcc03e6b8a0f635b24e2cc91ed1234c6688e6fb7.js:4941
     Shared Example Group: "React Component" called from ./spec/features/integration_spec.rb:31
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:376:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:27:in `block (3 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

Finished in 8.31 seconds (files took 10.52 seconds to load)
1 example, 1 failure

Failed examples:

rspec './spec/features/integration_spec.rb[1:1:1:1]' # Pages/Index All in one page Server Rendered/Cached React/Redux Component
</details>

### Hot Reload Issue
<details>
➜  ~/shakacode/react-on-rails/react_on_rails/spec/dummy (update-to-webpack-v2 u=) foreman start -f Procfile.hot                                                                                                                  [21:06:20]
21:06:30 rails.1               | started with pid 7264
21:06:30 hot-assets.1          | started with pid 7265
21:06:30 rails-server-assets.1 | started with pid 7266
21:06:30 hot-assets.1          | yarn run v0.20.3
21:06:30 rails-server-assets.1 | yarn run v0.20.3
21:06:31 rails-server-assets.1 | $ (cd client && yarn run build:dev:server --silent)
21:06:31 hot-assets.1          | $ (cd client && yarn run hot-assets)
21:06:32 hot-assets.1          | yarn run v0.20.3
21:06:32 rails-server-assets.1 | yarn run v0.20.3
21:06:32 hot-assets.1          | $ babel-node server-rails-hot.js
21:06:32 rails-server-assets.1 | $ webpack -w --config webpack.server.rails.build.config.js
21:06:33 rails-server-assets.1 |
21:06:33 rails-server-assets.1 | Webpack is watching the files…
21:06:33 rails-server-assets.1 |
21:06:35 rails.1               | => Booting Puma
21:06:35 rails.1               | => Rails 4.2.8 application starting in development on http://0.0.0.0:3000
21:06:35 rails.1               | => Run `rails server -h` for more startup options
21:06:35 rails.1               | => Ctrl-C to shutdown server
21:06:35 rails.1               | ================================================================================
21:06:35 rails.1               | Using caching = false
21:06:35 rails.1               | ================================================================================
21:06:36 hot-assets.1          | /Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/client/webpack.client.rails.hot.config.js:24
21:06:36 hot-assets.1          | config.module.loaders.push({
21:06:36 hot-assets.1          |                      ^
21:06:36 hot-assets.1          |
21:06:36 hot-assets.1          | TypeError: Cannot read property 'push' of undefined
21:06:36 hot-assets.1          |     at Object.<anonymous> (/Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/client/webpack.client.rails.hot.config.js:25:1)
21:06:36 hot-assets.1          |     at Module._compile (module.js:571:32)
21:06:36 hot-assets.1          |     at loader (/Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/client/node_modules/babel-register/lib/node.js:144:5)
21:06:36 hot-assets.1          |     at Object.require.extensions.(anonymous function) [as .js] (/Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/client/node_modules/babel-register/lib/node.js:154:7)
21:06:36 hot-assets.1          |     at Module.load (module.js:488:32)
21:06:36 hot-assets.1          |     at tryModuleLoad (module.js:447:12)
21:06:36 hot-assets.1          |     at Function.Module._load (module.js:439:3)
21:06:36 hot-assets.1          |     at Module.require (module.js:498:17)
21:06:36 hot-assets.1          |     at require (internal/module.js:20:19)
21:06:36 hot-assets.1          |     at Object.<anonymous> (/Users/justin/shakacode/react-on-rails/react_on_rails/spec/dummy/client/server-rails-hot.js:20:1)
21:06:36 hot-assets.1          | error Command failed with exit code 1.
21:06:36 hot-assets.1          | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
21:06:37 hot-assets.1          | error Command failed with exit code 1.
21:06:37 hot-assets.1          | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
21:06:37 hot-assets.1          | exited with code 1
21:06:37 system                | sending SIGTERM to all processes
21:06:37 rails.1               | Exiting
21:06:37 rails-server-assets.1 | terminated by SIGTERM
21:06:37 rails.1               | terminated by SIGTERM
</details>

**Comment by justin808** [comment]
Might be related: https://github.com/shakacode/react-webpack-rails-tutorial/issues/361

**Comment by alexfedoseev** [comment]
screen shot 2017-03-02 at 18 49 47

`config.module.rules.push`

**Comment by justin808** [comment]
Took out lots of errors!

Still got 6 left.

Please help!

<details><pre>
Failures:

  1) Pages/Index All in one page Server Rendered/Cached React Component Without Redux changes name in message according to input
     Failure/Error: visit root_path

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"a32319b9-08a8-46b9-8132-88848047fb57","name":"visit","args":["http://127.0.0.1:61221/",30]}
     Shared Example Group: "React Component" called from ./spec/features/integration_spec.rb:35
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:27:in `block (3 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  2) Pages/Index All in one page Simple Client Rendered Component same component with different props changes name in message according to input
     Failure/Error: visit root_path

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"3e03875f-515c-449e-a617-0ba5090b2da5","name":"visit","args":["http://127.0.0.1:61221/",30]}
     Shared Example Group: "React Component" called from ./spec/features/integration_spec.rb:42
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:27:in `block (3 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  3) Pages/Index All in one page Simple Component Without Redux changes name in message according to input
     Failure/Error: visit root_path

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"4651210f-e394-4916-b262-a044b8c436d1","name":"visit","args":["http://127.0.0.1:61221/",30]}
     Shared Example Group: "React Component" called from ./spec/features/integration_spec.rb:48
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:27:in `block (3 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  4) Code Splitting w/ Server Rendering loading an asyncronous route should not cause a client/server checksum mismatch
     Failure/Error: background { visit "/deferred_render_with_server_rendering/async_page" }

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"d6b357b7-74ab-4b20-b144-390a97c1ccf7","name":"visit","args":["http://127.0.0.1:61221/deferred_render_with_server_rendering/async_page",30]}
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:172:in `block (2 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  5) 2 react components, 1 store, server side, controller setup /server_side_hello_world_shared_store_controller Type in one component changes the other component
     Failure/Error: find("input").set new_text2

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"2075109b-1a9f-404d-8e39-ad9f49694d32","name":"set","args":[1,5,"Jane Smith"]}
     Shared Example Group: "React Component Shared Store" called from ./spec/features/integration_spec.rb:230
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:100:in `set'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/node.rb:17:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/node.rb:91:in `set'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/node/element.rb:108:in `block in set'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/node/base.rb:85:in `synchronize'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/node/element.rb:104:in `set'
     # ./spec/features/integration_spec.rb:206:in `block (4 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:291:in `within'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:202:in `block (3 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

  6) 2 react components, 1 store, server side, defer /server_side_hello_world_shared_store_defer Type in one component changes the other component
     Failure/Error: background { visit url }

     Capybara::Poltergeist::DeadClient:
       PhantomJS client died while processing {"id":"8bf1fd99-3813-4654-9913-e86630a9fab9","name":"visit","args":["http://127.0.0.1:61221/server_side_hello_world_shared_store_defer",30]}
     Shared Example Group: "React Component Shared Store" called from ./spec/features/integration_spec.rb:238
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/server.rb:38:in `send'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:369:in `command'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/browser.rb:35:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/poltergeist-1.12.0/lib/capybara/poltergeist/driver.rb:97:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/session.rb:233:in `visit'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/capybara-2.7.1/lib/capybara/dsl.rb:52:in `block (2 levels) in <module:DSL>'
     # ./spec/features/integration_spec.rb:190:in `block (2 levels) in <top (required)>'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:112:in `block in run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `loop'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:101:in `run'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rspec-retry-0.5.2/lib/rspec/retry.rb:30:in `block (2 levels) in setup'

Finished in 35.01 seconds (files took 5.03 seconds to load)
75 examples, 6 failures

Failed examples:

rspec './spec/features/integration_spec.rb[1:1:2:2]' # Pages/Index All in one page Server Rendered/Cached React Component Without Redux changes name in message according to input
rspec './spec/features/integration_spec.rb[1:1:3:3:2]' # Pages/Index All in one page Simple Client Rendered Component same component with different props changes name in message according to input
rspec './spec/features/integration_spec.rb[1:1:4:4]' # Pages/Index All in one page Simple Component Without Redux changes name in message according to input
rspec ./spec/features/integration_spec.rb:173 # Code Splitting w/ Server Rendering loading an asyncronous route should not cause a client/server checksum mismatch
rspec './spec/features/integration_spec.rb[16:1:1]' # 2 react components, 1 store, server side, controller setup /server_side_hello_world_shared_store_controller Type in one component changes the other component
rspec './spec/features/integration_spec.rb[18:1:1]' # 2 react components, 1 store, server side, defer /server_side_hello_world_shared_store_defer Type in one component changes the other component
</pre></details>

**Comment by justin808** [comment]
Can anybody tell me why travis is failing? https://travis-ci.org/shakacode/react_on_rails https://travis-ci.org/shakacode/react_on_rails/builds/208035037

but codeship is passing: https://app.codeship.com/projects/187011/builds/23158421

**Comment by justin808** [comment]
@robwise can you scan  these changes. I'd like to merge soon!

File: spec/dummy/client/package.json [hunk 6]
```diff
@@ -8,63 +8,62 @@
...
+    "eslint": "^3.16.1",
+    "eslint-config-shakacode": "^14.1.1",
     "eslint-plugin-import": "^2.2.0",
-    "eslint-plugin-jsx-a11y": "^3.0.2",
-    "eslint-plugin-react": "^6.8.0",
-    "fbjs": "^0.8.8",
-    "jsdom": "^9.9.1",
+    "eslint-plugin-jsx-a11y": "^4.0.0",
+    "eslint-plugin-react": "^6.10.0",
+    "fbjs": "^0.8.9",
```
Comment (alex35mil) [inline spec/dummy/client/package.json:56]: Why is this a direct dep?

Comment (justin808) [reply spec/dummy/client/package.json:56]: at some point, this was a peer dep...

**Pull request merged**